File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    if s < n:
        return [-1]
    
    # 방법
    # s(총합) // n(개수) 으로 이루어진 n개의 배열에 나머지를 반복문을 통해 더함
    answer = []
    a = s // n
    b = s % n
    answer = [a] * n
    for i in range(b):
        answer[i] += 1
        
    return sorted(answer)

File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.js
function solution(n, s) {
    if (s < n) {
        return [-1]
    }
    
    const a = parseInt(s / n)
    const b = s % n
    const answer = Array(n).fill(a)  // [a] * n in python
    
    for (let i=0; i< b; ++i){
        answer[i] += 1
    }
    
    // sort -> 항상 두 요소를 비교하여 작동. 결과가 0보다 작으면 a 먼저 표시, 크면 b 먼저 표시되는 원리
    answer.sort((a, b) => a - b)
    
    return answer
}
